tests/unit/test_ip6_address.cpp

```diff
@@ -134,8 +136,92 @@ void TestIp6AddressFromString(void)
     }
 }
 
+bool CheeckPrefix(const ot::Ip6::Address &aAddress, const uint8_t *aPrefix, uint8_t aPrefixLength)
+{
+    // Check the first aPrefixLength bits of aAddress to match the given aPrefix.
+
+    bool matches = true;
+
+    for (uint8_t bit = 0; bit < aPrefixLength; bit++)
+    {
+        uint8_t index = bit / CHAR_BIT;
+        uint8_t mask  = (0x80 >> (bit % CHAR_BIT));
+
+        if ((aAddress.mFields.m8[index] & mask) != (aPrefix[index] & mask))
+        {
+            matches = false;
+            break;
+        }
+    }
+
+    return matches;
+}
+
+bool CheckAddress(const ot::Ip6::Address &aAddress1, const ot::Ip6::Address &aAddress2, uint8_t aPrefixLength)
```

Comment:
Should be renamed to `CheckInterfaceId`? I think the name `CheckAddress` is misleading.